extensions/amp-next-page/0.1/amp-next-page.js

```diff
@@ -286,107 +86,20 @@ export class AmpNextPage extends AMP.BaseElement {
-   * end.
-   * @private
-   */
-  scrollHandler_() {
-    if (this.documentQueued_) {
-      return;
-    }
-
-    const viewportSize = this.viewport_.getSize();
-    const viewportBox =
-        layoutRectLtwh(0, 0, viewportSize.width, viewportSize.height);
-    this.viewport_.getClientRectAsync(this.element).then(elementBox => {
-      if (this.documentQueued_) {
-        return;
-      }
-
-      const prerenderHeight = PRERENDER_VIEWPORT_COUNT * viewportSize.height;
-      if (elementBox.bottom - viewportBox.bottom < prerenderHeight) {
-        this.documentQueued_ = true;
-        this.appendNextArticle_();
-      }
-    });
-  }
-
-  /**
-   * Handles updates from the {@code PositionObserver} indicating a change in
-   * the position of a recommendation unit in the viewport.
-   * @param {number} i Index of the documentRef this recommendation unit is
-   *     attached to.
-   * @param
-   * {!../../../src/service/position-observer/position-observer-worker.PositionInViewportEntryDef}
-   *     position Position of the current recommendation unit in the viewport.
-   */
-  positionUpdate_(i, position) {
-    // We're only interested when the recommendations exit the viewport
-    if (position.positionRect !== null) {
-      return;
-    }
-
-    if (position.relativePos === 'top') {
-      const documentRef = this.documentRefs_[i + 1];
-      this.triggerAnalyticsEvent_('amp-next-page-scroll',
-          documentRef.ampUrl, this.activeDocumentRef_.ampUrl);
-      this.setActiveDocument_(documentRef);
-    } else if (position.relativePos === 'bottom') {
-      const documentRef = this.documentRefs_[i];
-      this.triggerAnalyticsEvent_('amp-next-page-scroll-back',
-          documentRef.ampUrl, this.activeDocumentRef_.ampUrl);
-      this.setActiveDocument_(documentRef);
-    }
-  }
-
-  /**
-   * Sets the specified document as active, updating the document title and URL.
-   * @param {!DocumentRef} documentRef Reference to the document to set as
-   *     active.
-   */
-  setActiveDocument_(documentRef) {
-    const amp = documentRef.amp;
-    this.win.document.title = amp.title || '';
-    if (this.win.history.replaceState) {
-      const url = parseUrl(documentRef.ampUrl);
-      this.win.history.replaceState({}, amp.title, url.pathname);
-    }
-
-    this.activeDocumentRef_ = documentRef;
-
-    // TODO(peterjosling): Send request to viewer with title/URL
-    // TODO(emarchiori): Consider updating position fixed elements.
-  }
-
-  /**
-   * Wrapper around {@code triggerAnalyticsEvent}.
-   * @param {string} eventType
-   * @param {string} toURL The new url after the event.
-   * @param {string=} fromURL The old URL before the event.
-   */
-  triggerAnalyticsEvent_(eventType, toURL, fromURL) {
-    fromURL = fromURL || '';
-
-    const vars = {toURL, fromURL};
-    triggerAnalyticsEvent(this.element, eventType, vars);
+  appendPage_(element) {
+    this.element.appendChild(element);
```

Comment:
Ah I forgot to check for DOM mutations. We have an API to batch DOM mutations and it should be used here:

```js
this.mutateElement(() => this.element.appendChild(element));
```

Reply:
Putting it in a mutateElement breaks the rendering for some reason. When I tried it before it was complaining about something not being attached to the document, but now it's not logging any errors, but `position: fixed` elements aren't getting tagged as `fixedid`. Any ideas why?

Reply:
Might need to append the element before giving it to the MultiDocManager.

Reply:
Is it necessary to have the `mutateElement` given our `attachShadowDoc` runs in a `vsync`? Or should I remove the `vsync` and just use `mutateElement`?

Reply:
Remove the VSync and use mutateElement instead. VSync is being deprecated.

Reply:
Done, and added a second `mutateElement` from `Resources`. Thanks!